test_runner/src/main/kotlin/ftl/run/status/ExecutionStatusPrinter.kt

```diff
@@ -47,16 +47,28 @@ internal class MultiLinePrinter(
 
     private val output = LinkedHashMap<String, ExecutionStatus.View>()
     override fun invoke(changes: List<ExecutionStatus.Change>) {
-        repeat(output.size) {
-            print(ansi().cursorUpLine().eraseLine().toString())
-        }
-        output += changes.map { change ->
-            change.views.takeLast(1)
-        }.flatten().map { view ->
-            view.id to view
-        }
-        output.values.joinToString("\n").let(::println)
+        output.eraseBySize()
+        changes.mapChanges().compareToOutput(output).appendToOutput()?.print()
+    }
+
+    private fun LinkedHashMap<String, ExecutionStatus.View>.eraseBySize() = repeat(size) {
+        print(ansi().cursorUpLine().eraseLine().toString())
+    }
+
+    private fun List<ExecutionStatus.Change>.mapChanges() = map { change ->
+        change.views.takeLast(1)
+    }.flatten().map { view ->
+        view.id to view
     }
+
+    private fun List<Pair<String, ExecutionStatus.View>>.compareToOutput(out: LinkedHashMap<String, ExecutionStatus.View>) =
+        if (out.isEmpty()) this
+        else filter { out[it.first] != it.second }
+
+    private fun List<Pair<String, ExecutionStatus.View>>.appendToOutput() = takeIf { isNotEmpty() }.also { output += this }
```

Comment:
how about specify that this is conditional append?
`appendToOutput` -> `appendToOutputIfNotEmpty` ?